src/modules/persistence/services/bulk.updates.service.ts

```diff
@@ -75,6 +76,54 @@ export class BulkUpdatesService {
         return this.convertUpdatesToMongoBulkOperations();
     }
 
+    getDbUpdateOperations(
+        stateChangesMap: Map<string, PairStateChanges>,
+    ): MongoBulkOperations {
+        for (const [address, stateChanges] of stateChangesMap.entries()) {
+            this.updatePairState(address, stateChanges);
+            this.updatePairReservesAndPrices(address, stateChanges);
+        }
+
+        if (this.pricesRecomputeNeeded) {
+            this.updateTokensDerivedEgldAndUsdPrices();
+            this.updateValuesUSD();
+        }
+
+        return this.convertUpdatesToMongoBulkOperations();
+    }
+
+    private updatePairState(
+        address: string,
+        stateChanges: PairStateChanges,
+    ): void {
+        const trackedFields = [
+            TrackedPairFields.state,
+            TrackedPairFields.totalFeePercent,
+            TrackedPairFields.specialFeePercent,
+        ];
+
+        const rawPair: Partial<PairModel> = {};
+
+        for (const field of Object.keys(stateChanges)) {
+            if (!trackedFields.includes(field as TrackedPairFields)) {
```

Comment:
this if statement can be moved outside of the for loop and early exit if no tracked fields are in state changes